cmd/xl-v1-metadata.go

```diff
@@ -173,7 +176,8 @@ func newXLMetaV1(object string, dataBlocks, parityBlocks int) (xlMeta xlMetaV1)
 // IsValid - tells if the format is sane by validating the version
 // string and format style.
 func (m xlMetaV1) IsValid() bool {
```

Comment:
> One caveat is that etag will be missing if users run older binaries with newer file..

exactly this is the reason it should have failed. All the clients which depend on etag will be affected. There can be more dangerous consequences which we can not predict at this time. You can confirm with AB about this if you'd like.